File: ats-ness-front-prod/src/app/demo/models/Prospection.ts
export interface Prospection {
    id: number;
    nbrAppels: number;
    nbrMails: number;
    datePremiereAction: string;  // Utilisation de string pour la date (format ISO 8601)
    dateDerniereAction: string;  // Utilisation de string pour la date (format ISO 8601)
    nbrJoursDepuisDerniereAction: number;
    ajoutLinkedIn: boolean;
    dateajoutLinkedIn: string;  // Utilisation de string pour la date (format ISO 8601)
    linkedInAccepte: boolean;
    dateAcceptationLinkedIn: string;  // Utilisation de string pour la date (format ISO 8601)
    mailLinkedInEnvoye: boolean;
    dateEnvoiMailLinkedIn: string;  // Utilisation de string pour la date (format ISO 8601)
}

File: ats-ness-back-prod/src/main/java/nesstechnologies/fr/applicationesnbackend/entities/Prospection.java
package nesstechnologies.fr.applicationesnbackend.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDateTime;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
public class Prospection implements Serializable { @Id
@GeneratedValue(strategy = GenerationType.IDENTITY)
private Integer id;
    @JsonProperty("nbrAppels")
    private Integer nbrAppels;
    @JsonProperty("nbrMails")
    private Integer nbrMails;
    @JsonProperty("datePremiereAction")
    private LocalDateTime datePremiereAction;
    @JsonProperty("dateDerniereAction")
    private LocalDateTime dateDerniereAction;
    @JsonProperty("nbrJoursDepuisDerniereAction")
    private Integer nbrJoursDepuisDerniereAction;
    @JsonProperty("ajoutLinkedIn")
    private boolean ajoutLinkedIn;
    @JsonProperty("dateajoutLinkedIn")
    private LocalDateTime dateajoutLinkedIn;
    @JsonProperty("linkedInAccepte")
    private boolean LinkedqInAccepte;
    @JsonProperty("dateAcceptationLinkedIn")
    private LocalDateTime dateAcceptationLinkedIn;
    @JsonProperty("mailLinkedInEnvoye")
    private boolean mailLinkedInEnvoye;
    @JsonProperty("dateEnvoiMailLinkedIn")
    private LocalDateTime dateEnvoiMailLinkedIn;

}
